pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -194,6 +209,9 @@ func (c *ClientCache) FederationClientSetForVersion(version *schema.GroupVersion
 }
 
 func (c *ClientCache) FederationClientForVersion(version *schema.GroupVersion) (*restclient.RESTClient, error) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
 	fedClientSet, err := c.FederationClientSetForVersion(version)
```

Comment:
@adamreese Please change L215 from fedClientSet, err := c.FederationClientSetForVersion(version) to fedClientSet, err := c.federationClientSetForVersion(version), and update the PR? 

Thanks!